spec/core/ics-003-connection-semantics/UPGRADES.md

```diff
@@ -0,0 +1,407 @@
+  state: ConnectionState
+  counterpartyConnectionIdentifier: Identifier
+  counterpartyPrefix: CommitmentPrefix
+  clientIdentifier: Identifier
+  counterpartyClientIdentifier: Identifier
+  version: string | []string
+  delayPeriodTime: uint64
+  delayPeriodBlocks: uint64
+}
+```
+
+The desired property that the connection upgrade protocol may not modify the underlying clients or connection identifiers, means that only some fields are upgradable by the upgrade protocol.
+
+- `state`: The state is specified by the handshake steps of the upgrade protocol.
+
+CAN BE MODIFIED:
+- `counterpartyPrefix`: The prefix MAY be modified in the upgrade protocol. The counterparty must accept the new proposed prefix value, or it must return an error during the upgrade handshake.
+- `version`: The version MAY be modified by the upgrade protocol. The same version negotiation that happens in the initial connection handshake can be employed for the upgrade handshake.
+- `delayPeriodTime`: The delay period MAY be modified by the upgrade protocol. The counterparty MUST accept the new proposed value or return an error during the upgrade handshake.
+- `delayPeriodBlocks`: The delay period MAY be modified by the upgrade protocol. The counterparty MUST accept the new proposed value or return an error during the upgrade handshake.
+
+CANNOT BE MODIFIED:
+- `counterpartyConnectionIdentifier`: The counterparty connection identifier CAN NOT be modified by the upgrade protocol.
+- `clientIdentifier`: The client identifier CAN NOT be modified by the upgrade protocol
+- `counterpartyClientIdentifier`: The counterparty client identifier CAN NOT be modified by the upgrade protocol
+
+NOTE: If the upgrade adds any fields to the `ConnectionEnd` these are by default modifiable, and can be arbitrarily chosen by an Actor (e.g. chain governance) which has permission to initiate the upgrade.
+
+```typescript
+interface UpgradeStatus {
+    state: ConnectionState
+    timeoutHeight: Height
+    timeoutTimestamp: uint64
+}
+```
+
+- UpgradeStatus contains the state of the upgrade. If upgrade is successful it will contain the `ConnectionState` of the upgrading connection on the executing chain. If the executing chain wants to abort the upgrade, it will restore its previous connection under its connection path with state `OPEN` and write an `UpgradeStatus` with state `UPGRADE_ERR`.
+- `timeoutHeight`: Timeout height indicates the height at which the counterparty must no longer proceed with the upgrade handshake. The chains will then preserve their original connection and the upgrade handshake is aborted.
+- `timeoutTimestamp`: Timeout timestamp indicates the time on the counterparty at which the counterparty must no longer proceed with the upgrade handshake. The chains will then preserve their original connection and the upgrade handshake is aborted.
+
+```typescript
+interface UpgradeConnectionState {
+    connection: ConnectionEnd
+    timeoutHeight: Height
+    timeoutTimestamp: uint64
```

Comment:
You're right. I've removed `UpgradeConnectionState` and `UpgradeStatus`. Replaced with `UpgradeTimeout` and I believe the protocol is much cleaner.

Leaving the struct fields as is, since they are the same semantically as the packet timeouts. But changed the naming in the OpenInit function